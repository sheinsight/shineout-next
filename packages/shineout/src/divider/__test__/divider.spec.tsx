import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Divider from '..';
import mountTest from '../../tests/mountTest';
import { classLengthTest } from '../../tests/structureTest';
import {
  baseTest,
  childrenTest,
  classContentTest,
  createClassName,
  displayTest,
  snapshotTest,
  textContentTest,
} from '../../tests/utils';
import DividerBase from '../__example__/01-base';
import DividerText from '../__example__/02-text';
import DividerVertical from '../__example__/03-vertical';

const SO_PREFIX = 'divider';
const originClasses = ['wrapper', 'innerText'];
const originItemClasses = [
  'vertical',
  'horizontal',
  'withText',
  'withTextCenter',
  'withTextLeft',
  'withTextRight',
];

const {
  wrapper,
  vertical,
  horizontal,
  withText,
  withTextCenter,
  withTextLeft,
  withTextRight,
  innerText,
} = createClassName(SO_PREFIX, originClasses, originItemClasses);

const classNamesByContextMaps = {
  Center: withTextCenter,
  Left: withTextLeft,
  Right: withTextRight,
};

afterEach(cleanup);
mountTest(<Divider />);
describe('Divider[Base]', () => {
  displayTest(Divider, 'ShineoutDivider');
  baseTest(Divider, wrapper);
  childrenTest(Divider, wrapper);
  snapshotTest(<DividerBase />);
  snapshotTest(<DividerText />, 'about text');
  snapshotTest(<DividerVertical />, 'about vertical');
  test('should render default', () => {
    const { container } = render(<Divider />);
    classContentTest(container.querySelector(wrapper)!, horizontal);
  });
  test('should render when set mode is vertical', () => {
    const { container } = render(<DividerVertical />);
    const dividers = container.querySelectorAll(wrapper)!;
    dividers.forEach((item) => {
      classContentTest(item, vertical);
    });
  });
  test('should render when set vertical and children', () => {
    const { container } = render(<Divider mode='vertical'>demo</Divider>);
    const divider = container.querySelector(wrapper)!;
    classContentTest(divider, vertical);
    expect(divider.textContent).toBe('');
  });
  test('should render default when set children', () => {
    const { container } = render(<Divider>Center</Divider>);
    const divider = container.querySelector(wrapper)!;
    classContentTest(divider, withText);
    classLengthTest(divider, innerText, 1);
    classContentTest(divider, withTextCenter);
    textContentTest(divider, 'Center');
  });
  test('should render when set orientation', () => {
    const { container } = render(<DividerText />);
    const dividers = container.querySelectorAll(wrapper)!;
    dividers.forEach((item) => {
      classContentTest(item, withText);
      classLengthTest(item, innerText, 1);
      classContentTest(
        item,
        classNamesByContextMaps[item.textContent as 'Center' | 'Left' | 'Right'],
      );
    });
  });
});
