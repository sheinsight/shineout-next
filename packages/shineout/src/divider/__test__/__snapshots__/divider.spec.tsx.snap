// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider[Base] should render correctly  1`] = `
<div
  style="font-size: 12px;"
>
  <p>
    A design is a plan or specification for the construction of an object.
  </p>
  <div
    class="so-divider-wrapper so-divider-horizontal"
  />
  <p>
    A design is a plan or specification for the construction of an object.
  </p>
</div>
`;

exports[`Divider[Base] should render correctly about text 1`] = `
<div
  style="font-size: 12px;"
>
  <p>
    A design is a plan or specification for the construction of an object.
  </p>
  <div
    class="so-divider-wrapper so-divider-horizontal so-divider-with-text so-divider-with-text-center"
  >
    <span
      class="so-divider-inner-text"
    >
      Center
    </span>
  </div>
  <p>
    A design is a plan or specification for the construction of an object.
  </p>
  <div
    class="so-divider-wrapper so-divider-horizontal so-divider-with-text so-divider-with-text-left"
  >
    <span
      class="so-divider-inner-text"
    >
      Left
    </span>
  </div>
  <p>
    A design is a plan or specification for the construction of an object.
  </p>
  <div
    class="so-divider-wrapper so-divider-horizontal so-divider-with-text so-divider-with-text-right"
  >
    <span
      class="so-divider-inner-text"
    >
      Right
    </span>
  </div>
  <p>
    A design is a plan or specification for the construction of an object.
  </p>
</div>
`;

exports[`Divider[Base] should render correctly about vertical 1`] = `
<div
  style="font-size: 12px;"
>
  <span>
    Left
  </span>
  <div
    class="so-divider-wrapper so-divider-vertical"
  />
  <span>
    Center
  </span>
  <div
    class="so-divider-wrapper so-divider-vertical"
  />
  <span>
    Right
  </span>
</div>
`;
